src/audiobuffer/buffer.h

```diff
@@ -0,0 +1,109 @@
+#ifndef CIRCULAR_BUFFER_H
+#define CIRCULAR_BUFFER_H
+
+// #ifndef _IS_WIN_
+// #include <algorithm>
+// #endif
+
+#include <vector>
+#include <cstdint>
+#include <algorithm>
+#include <iostream>
+#include <cstring>
+
+enum Endianness // TODO?
+{
+    BUFFER_LITTLE_ENDIAN,
+    BUFFER_BIG_ENDIAN
+};
+
+class Buffer
+{
+public:
+    std::vector<int8_t> buffer; // Buffer that stores int8_t data
+
+private:
+    size_t maxBytes; // Maximum capacity in bytes
+
+public:
+    // Constructor that accepts the maxBytes parameter
+    Buffer() : maxBytes(0) {}
+
+    ~Buffer()
+    {
+        clear();
+    }
+
+    void setSizeInBytes(size_t newBytes)
+    {
+        maxBytes = newBytes;
+    }
+
+    // Overload for int8_t data, converting it to normalized float and adding its bytes to the buffer
+    // Return the number of floats written.
+    size_t addData(const int8_t* data, size_t numSamples) {
```

Comment:
Do we really need all these overloads? These are basically internal methods, right? Maybe, to keep the API as minimal as possible, it would be better to just keep a single method?

Reply:
Done.